diff --git ./src/main/java/org/jboss/examples/ticketmonster/rest/PerformanceEndpoint.java ./src/main/java/org/jboss/examples/ticketmonster/rest/PerformanceEndpoint.java
index 4d5c0b3..3f0e6ba 100644
--- ./src/main/java/org/jboss/examples/ticketmonster/rest/PerformanceEndpoint.java
+++ ./src/main/java/org/jboss/examples/ticketmonster/rest/PerformanceEndpoint.java
@@ -9,6 +9,10 @@ import javax.persistence.NoResultException;
 import javax.persistence.OptimisticLockException;
 import javax.persistence.PersistenceContext;
 import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
 import javax.ws.rs.Consumes;
 import javax.ws.rs.DELETE;
 import javax.ws.rs.GET;
@@ -21,8 +25,12 @@ import javax.ws.rs.QueryParam;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.Response.Status;
 import javax.ws.rs.core.UriBuilder;
+
 import org.jboss.examples.ticketmonster.rest.dto.PerformanceDTO;
+import org.jboss.examples.ticketmonster.model.Booking;
 import org.jboss.examples.ticketmonster.model.Performance;
+import org.jboss.examples.ticketmonster.model.SectionAllocation;
+import org.jboss.examples.ticketmonster.model.Show;
 
 /**
  * 
@@ -52,6 +60,18 @@ public class PerformanceEndpoint
       {
          return Response.status(Status.NOT_FOUND).build();
       }
+      Show show = entity.getShow();
+      show.getPerformances().remove(entity);
+      entity.setShow(null);
+      this.em.merge(show);
+      List<SectionAllocation> sectionAllocations = findSectionAllocationsByPerformance(entity);
+      for(SectionAllocation sectionAllocation: sectionAllocations) {
+         this.em.remove(sectionAllocation);
+      }
+      List<Booking> bookings = findBookingsByPerformance(entity);
+      for(Booking booking: bookings) {
+         this.em.remove(booking);
+      }
       em.remove(entity);
       return Response.noContent().build();
    }
@@ -130,4 +150,27 @@ public class PerformanceEndpoint
       }
       return Response.noContent().build();
    }
+
+   public List<SectionAllocation> findSectionAllocationsByPerformance(Performance performance)
+   {
+      CriteriaQuery<SectionAllocation> criteria = this.em
+            .getCriteriaBuilder().createQuery(SectionAllocation.class);
+      Root<SectionAllocation> from = criteria.from(SectionAllocation.class);
+      CriteriaBuilder builder = this.em.getCriteriaBuilder();
+      Predicate performanceIsSame = builder.equal(from.get("performance"), performance);
+      return this.em.createQuery(
+            criteria.select(from).where(performanceIsSame)).getResultList();
+   }
+
+   public List<Booking> findBookingsByPerformance(Performance performance)
+   {
+      CriteriaQuery<Booking> criteria = this.em
+            .getCriteriaBuilder().createQuery(Booking.class);
+      Root<Booking> from = criteria.from(Booking.class);
+      CriteriaBuilder builder = this.em.getCriteriaBuilder();
+      Predicate performanceIsSame = builder.equal(from.get("performance"), performance);
+      return this.em.createQuery(
+            criteria.select(from).where(performanceIsSame)).getResultList();
+   }
+
 }
diff --git ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/NestedPerformanceDTO.java ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/NestedPerformanceDTO.java
index 1fb77f9..421c55f 100644
--- ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/NestedPerformanceDTO.java
+++ ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/NestedPerformanceDTO.java
@@ -11,6 +11,7 @@ public class NestedPerformanceDTO implements Serializable
 
    private Long id;
    private Date date;
+   private String displayTitle;
 
    public NestedPerformanceDTO()
    {
@@ -22,6 +23,7 @@ public class NestedPerformanceDTO implements Serializable
       {
          this.id = entity.getId();
          this.date = entity.getDate();
+         this.displayTitle = entity.toString();
       }
    }
 
@@ -72,4 +74,9 @@ public class NestedPerformanceDTO implements Serializable
    {
       this.date = date;
    }
-}
\ No newline at end of file
+
+   public String getDisplayTitle()
+   {
+      return this.displayTitle;
+   }
+}
diff --git ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/NestedShowDTO.java ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/NestedShowDTO.java
index c5207cf..05f69e1 100644
--- ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/NestedShowDTO.java
+++ ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/NestedShowDTO.java
@@ -9,6 +9,7 @@ public class NestedShowDTO implements Serializable
 {
 
    private Long id;
+   private String displayTitle;
 
    public NestedShowDTO()
    {
@@ -19,6 +20,7 @@ public class NestedShowDTO implements Serializable
       if (entity != null)
       {
          this.id = entity.getId();
+         this.displayTitle = entity.toString();
       }
    }
 
@@ -57,4 +59,9 @@ public class NestedShowDTO implements Serializable
    {
       this.id = id;
    }
-}
\ No newline at end of file
+
+   public String getDisplayTitle()
+   {
+      return this.displayTitle;
+   }
+}
diff --git ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/NestedTicketPriceDTO.java ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/NestedTicketPriceDTO.java
index 54d7f7d..0165de9 100644
--- ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/NestedTicketPriceDTO.java
+++ ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/NestedTicketPriceDTO.java
@@ -10,6 +10,7 @@ public class NestedTicketPriceDTO implements Serializable
 
    private Long id;
    private float price;
+   private String displayTitle;
 
    public NestedTicketPriceDTO()
    {
@@ -21,6 +22,7 @@ public class NestedTicketPriceDTO implements Serializable
       {
          this.id = entity.getId();
          this.price = entity.getPrice();
+         this.displayTitle = entity.toString();
       }
    }
 
@@ -71,4 +73,9 @@ public class NestedTicketPriceDTO implements Serializable
    {
       this.price = price;
    }
-}
\ No newline at end of file
+
+   public String getDisplayTitle()
+   {
+      return this.displayTitle;
+   }
+}
diff --git ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/PerformanceDTO.java ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/PerformanceDTO.java
index 83d6bdf..b4a2e1f 100644
--- ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/PerformanceDTO.java
+++ ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/PerformanceDTO.java
@@ -14,6 +14,7 @@ public class PerformanceDTO implements Serializable
    private Long id;
    private NestedShowDTO show;
    private Date date;
+   private String displayTitle;
 
    public PerformanceDTO()
    {
@@ -26,6 +27,7 @@ public class PerformanceDTO implements Serializable
          this.id = entity.getId();
          this.show = new NestedShowDTO(entity.getShow());
          this.date = entity.getDate();
+         this.displayTitle = entity.toString();
       }
    }
 
@@ -73,4 +75,9 @@ public class PerformanceDTO implements Serializable
    {
       this.date = date;
    }
-}
\ No newline at end of file
+
+   public String getDisplayTitle()
+   {
+      return this.displayTitle;
+   }
+}
diff --git ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/ShowDTO.java ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/ShowDTO.java
index 37b4b5e..2e4ceda 100644
--- ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/ShowDTO.java
+++ ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/ShowDTO.java
@@ -1,17 +1,32 @@
 package org.jboss.examples.ticketmonster.rest.dto;
 
 import java.io.Serializable;
+
+import org.jboss.examples.ticketmonster.model.Booking;
+import org.jboss.examples.ticketmonster.model.SectionAllocation;
 import org.jboss.examples.ticketmonster.model.Show;
+
 import javax.persistence.EntityManager;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
 import org.jboss.examples.ticketmonster.rest.dto.NestedEventDTO;
+
+import java.util.List;
 import java.util.Set;
 import java.util.HashSet;
+
 import org.jboss.examples.ticketmonster.rest.dto.NestedPerformanceDTO;
 import org.jboss.examples.ticketmonster.model.Performance;
+
 import java.util.Iterator;
+
 import org.jboss.examples.ticketmonster.rest.dto.NestedVenueDTO;
 import org.jboss.examples.ticketmonster.rest.dto.NestedTicketPriceDTO;
 import org.jboss.examples.ticketmonster.model.TicketPrice;
+
 import javax.xml.bind.annotation.XmlRootElement;
 
 @XmlRootElement
@@ -23,6 +38,7 @@ public class ShowDTO implements Serializable
    private Set<NestedPerformanceDTO> performances = new HashSet<NestedPerformanceDTO>();
    private NestedVenueDTO venue;
    private Set<NestedTicketPriceDTO> ticketPrices = new HashSet<NestedTicketPriceDTO>();
+   private String displayTitle;
 
    public ShowDTO()
    {
@@ -49,6 +65,7 @@ public class ShowDTO implements Serializable
             TicketPrice element = iterTicketPrices.next();
             this.ticketPrices.add(new NestedTicketPriceDTO(element));
          }
+         this.displayTitle = entity.toString();
       }
    }
 
@@ -83,6 +100,17 @@ public class ShowDTO implements Serializable
          if (found == false)
          {
             iterPerformances.remove();
+            List<SectionAllocation> sectionAllocations = findSectionAllocationsByPerformance(performance, em);
+            for(SectionAllocation sectionAllocation: sectionAllocations)
+            {
+               em.remove(sectionAllocation);
+            }
+            List<Booking> bookings = findBookingsByPerformance(performance, em);
+            for(Booking booking: bookings)
+            {
+               em.remove(booking);
+            }
+            em.remove(performance);
          }
       }
       Iterator<NestedPerformanceDTO> iterDtoPerformances = this
@@ -180,6 +208,27 @@ public class ShowDTO implements Serializable
       return entity;
    }
 
+   public List<SectionAllocation> findSectionAllocationsByPerformance(Performance performance, EntityManager em)
+   {
+      CriteriaQuery<SectionAllocation> criteria = em
+             .getCriteriaBuilder().createQuery(SectionAllocation.class);
+      Root<SectionAllocation> from = criteria.from(SectionAllocation.class);
+      CriteriaBuilder builder = em.getCriteriaBuilder();
+      Predicate performanceIsSame = builder.equal(from.get("performance"), performance);
+      return em.createQuery(
+             criteria.select(from).where(performanceIsSame)).getResultList();
+   }
+   
+   public List<Booking> findBookingsByPerformance(Performance performance, EntityManager em)
+   {
+      CriteriaQuery<Booking> criteria = em.getCriteriaBuilder().createQuery(Booking.class);
+      Root<Booking> from = criteria.from(Booking.class);
+      CriteriaBuilder builder = em.getCriteriaBuilder();
+      Predicate performanceIsSame = builder.equal(from.get("performance"), performance);
+      return em.createQuery(
+             criteria.select(from).where(performanceIsSame)).getResultList();
+   }
+
    public Long getId()
    {
       return this.id;
@@ -229,4 +278,9 @@ public class ShowDTO implements Serializable
    {
       this.ticketPrices = ticketPrices;
    }
-}
\ No newline at end of file
+
+   public String getDisplayTitle()
+   {
+      return this.displayTitle;
+   }
+}
diff --git ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/TicketPriceDTO.java ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/TicketPriceDTO.java
index 455f614..bce2cb3 100644
--- ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/TicketPriceDTO.java
+++ ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/TicketPriceDTO.java
@@ -17,6 +17,7 @@ public class TicketPriceDTO implements Serializable
    private NestedSectionDTO section;
    private NestedTicketCategoryDTO ticketCategory;
    private float price;
+   private String displayTitle;
 
    public TicketPriceDTO()
    {
@@ -32,6 +33,7 @@ public class TicketPriceDTO implements Serializable
          this.ticketCategory = new NestedTicketCategoryDTO(
                entity.getTicketCategory());
          this.price = entity.getPrice();
+         this.displayTitle = entity.toString();
       }
    }
 
@@ -108,4 +110,9 @@ public class TicketPriceDTO implements Serializable
    {
       this.price = price;
    }
-}
\ No newline at end of file
+
+   public String getDisplayTitle()
+   {
+      return this.displayTitle;
+   }
+}
diff --git ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/VenueDTO.java ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/VenueDTO.java
index 2412c30..8048ddf 100644
--- ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/VenueDTO.java
+++ ./src/main/java/org/jboss/examples/ticketmonster/rest/dto/VenueDTO.java
@@ -1,15 +1,30 @@
 package org.jboss.examples.ticketmonster.rest.dto;
 
 import java.io.Serializable;
+
+import org.jboss.examples.ticketmonster.model.SectionAllocation;
+import org.jboss.examples.ticketmonster.model.Show;
+import org.jboss.examples.ticketmonster.model.TicketPrice;
 import org.jboss.examples.ticketmonster.model.Venue;
+
 import javax.persistence.EntityManager;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
 import org.jboss.examples.ticketmonster.rest.dto.AddressDTO;
 import org.jboss.examples.ticketmonster.rest.dto.NestedMediaItemDTO;
+
+import java.util.List;
 import java.util.Set;
 import java.util.HashSet;
+
 import org.jboss.examples.ticketmonster.rest.dto.NestedSectionDTO;
 import org.jboss.examples.ticketmonster.model.Section;
+
 import java.util.Iterator;
+
 import javax.xml.bind.annotation.XmlRootElement;
 
 @XmlRootElement
@@ -83,6 +98,19 @@ public class VenueDTO implements Serializable
          if (found == false)
          {
             iterSections.remove();
+            List<SectionAllocation> sectionAllocations = findSectionAllocationBySection(section, em);
+	        for(SectionAllocation sectionAllocation: sectionAllocations)
+            {
+               em.remove(sectionAllocation);
+            }
+            List<TicketPrice> ticketPrices = findTicketPricesBySection(section, em);
+            for(TicketPrice ticketPrice: ticketPrices)
+            {
+               Show show = ticketPrice.getShow();
+               show.getTicketPrices().remove(ticketPrice);
+               em.remove(ticketPrice);
+            }
+            em.remove(section);
          }
       }
       Iterator<NestedSectionDTO> iterDtoSections = this.getSections()
@@ -122,6 +150,28 @@ public class VenueDTO implements Serializable
       return entity;
    }
 
+   public List<SectionAllocation> findSectionAllocationBySection(Section section, EntityManager em)
+   {
+      CriteriaQuery<SectionAllocation> criteria = em
+             .getCriteriaBuilder().createQuery(SectionAllocation.class);
+      Root<SectionAllocation> from = criteria.from(SectionAllocation.class);
+      CriteriaBuilder builder = em.getCriteriaBuilder();
+      Predicate sectionIsSame = builder.equal(from.get("section"), section);
+      return em.createQuery(
+             criteria.select(from).where(sectionIsSame)).getResultList();
+   }
+   
+   public List<TicketPrice> findTicketPricesBySection(Section section, EntityManager em)
+   {
+      CriteriaQuery<TicketPrice> criteria = em
+             .getCriteriaBuilder().createQuery(TicketPrice.class);
+      Root<TicketPrice> from = criteria.from(TicketPrice.class);
+      CriteriaBuilder builder = em.getCriteriaBuilder();
+      Predicate sectionIsSame = builder.equal(from.get("section"), section);
+      return em.createQuery(
+             criteria.select(from).where(sectionIsSame)).getResultList();
+   }
+
    public Long getId()
    {
       return this.id;
@@ -191,4 +241,4 @@ public class VenueDTO implements Serializable
    {
       this.capacity = capacity;
    }
-}
\ No newline at end of file
+}
diff --git ./src/main/webapp/admin/scripts/controllers/editBookingController.js ./src/main/webapp/admin/scripts/controllers/editBookingController.js
index 91ddde7..6233ab4 100644
--- ./src/main/webapp/admin/scripts/controllers/editBookingController.js
+++ ./src/main/webapp/admin/scripts/controllers/editBookingController.js
@@ -37,7 +37,7 @@ angular.module('ticketmonster').controller('EditBookingController', function($sc
                     };
                     var labelObject = {
                         value : item.id,
-                        text : item.date
+                        text : item.displayTitle
                     };
                     if($scope.booking.performance && item.id == $scope.booking.performance.id) {
                         $scope.performanceSelection = labelObject;
@@ -103,4 +103,4 @@ angular.module('ticketmonster').controller('EditBookingController', function($sc
     });
     
     $scope.get();
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/scripts/controllers/editEventController.js ./src/main/webapp/admin/scripts/controllers/editEventController.js
index 8bb871b..553bdf2 100644
--- ./src/main/webapp/admin/scripts/controllers/editEventController.js
+++ ./src/main/webapp/admin/scripts/controllers/editEventController.js
@@ -16,7 +16,7 @@ angular.module('ticketmonster').controller('EditEventController', function($scop
                     };
                     var labelObject = {
                         value : item.id,
-                        text : item.mediaType
+                        text : item.url
                     };
                     if($scope.event.mediaItem && item.id == $scope.event.mediaItem.id) {
                         $scope.mediaItemSelection = labelObject;
@@ -94,4 +94,4 @@ angular.module('ticketmonster').controller('EditEventController', function($scop
     });
     
     $scope.get();
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/scripts/controllers/editPerformanceController.js ./src/main/webapp/admin/scripts/controllers/editPerformanceController.js
index fc93d6a..98e21fc 100644
--- ./src/main/webapp/admin/scripts/controllers/editPerformanceController.js
+++ ./src/main/webapp/admin/scripts/controllers/editPerformanceController.js
@@ -16,7 +16,7 @@ angular.module('ticketmonster').controller('EditPerformanceController', function
                     };
                     var labelObject = {
                         value : item.id,
-                        text : item.id
+                        text : item.displayTitle
                     };
                     if($scope.performance.show && item.id == $scope.performance.show.id) {
                         $scope.showSelection = labelObject;
@@ -71,4 +71,4 @@ angular.module('ticketmonster').controller('EditPerformanceController', function
     });
     
     $scope.get();
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/scripts/controllers/editSectionAllocationController.js ./src/main/webapp/admin/scripts/controllers/editSectionAllocationController.js
index 763fa18..734baae 100644
--- ./src/main/webapp/admin/scripts/controllers/editSectionAllocationController.js
+++ ./src/main/webapp/admin/scripts/controllers/editSectionAllocationController.js
@@ -16,7 +16,7 @@ angular.module('ticketmonster').controller('EditSectionAllocationController', fu
                     };
                     var labelObject = {
                         value : item.id,
-                        text : item.date
+                        text : item.displayTitle
                     };
                     if($scope.sectionAllocation.performance && item.id == $scope.sectionAllocation.performance.id) {
                         $scope.performanceSelection = labelObject;
@@ -94,4 +94,4 @@ angular.module('ticketmonster').controller('EditSectionAllocationController', fu
     });
     
     $scope.get();
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/scripts/controllers/editShowController.js ./src/main/webapp/admin/scripts/controllers/editShowController.js
index 04d537a..e863587 100644
--- ./src/main/webapp/admin/scripts/controllers/editShowController.js
+++ ./src/main/webapp/admin/scripts/controllers/editShowController.js
@@ -33,7 +33,7 @@ angular.module('ticketmonster').controller('EditShowController', function($scope
                     };
                     var labelObject = {
                         value : item.id,
-                        text : item.date
+                        text : item.displayTitle
                     };
                     if($scope.show.performances){
                         $.each($scope.show.performances, function(idx, element) {
@@ -71,7 +71,7 @@ angular.module('ticketmonster').controller('EditShowController', function($scope
                     };
                     var labelObject = {
                         value : item.id,
-                        text : item.price
+                        text : item.displayTitle
                     };
                     if($scope.show.ticketPrices){
                         $.each($scope.show.ticketPrices, function(idx, element) {
@@ -158,4 +158,4 @@ angular.module('ticketmonster').controller('EditShowController', function($scope
     });
     
     $scope.get();
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/scripts/controllers/editTicketPriceController.js ./src/main/webapp/admin/scripts/controllers/editTicketPriceController.js
index 532ab5e..0f5446d 100644
--- ./src/main/webapp/admin/scripts/controllers/editTicketPriceController.js
+++ ./src/main/webapp/admin/scripts/controllers/editTicketPriceController.js
@@ -16,7 +16,7 @@ angular.module('ticketmonster').controller('EditTicketPriceController', function
                     };
                     var labelObject = {
                         value : item.id,
-                        text : item.id
+                        text : item.displayTitle
                     };
                     if($scope.ticketPrice.show && item.id == $scope.ticketPrice.show.id) {
                         $scope.showSelection = labelObject;
@@ -101,6 +101,27 @@ angular.module('ticketmonster').controller('EditTicketPriceController', function
         if (typeof selection != 'undefined') {
             $scope.ticketPrice.show = {};
             $scope.ticketPrice.show.id = selection.value;
+            if($scope.showList) {
+                var selectedShow = $.map($scope.showList, function(show) {
+                    if(show.id === $scope.ticketPrice.show.id) {
+                        return show;
+                    }
+                });
+                if(selectedShow.length > 0) {
+                    selectedShow = selectedShow[0];
+                }
+            }
+            if($scope.sectionList && selectedShow) {
+                var venueSectionList = $.map($scope.sectionList, function(section) {
+                    if(selectedShow.venue.id === section.venue.id) {
+                        return ( {
+                            value : section.id,
+                            text : section.name
+                        });
+                    }
+                });
+                $scope.sectionSelectionList = venueSectionList;
+            }
         }
     });
     $scope.$watch("sectionSelection", function(selection) {
@@ -117,4 +138,4 @@ angular.module('ticketmonster').controller('EditTicketPriceController', function
     });
     
     $scope.get();
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/scripts/controllers/editVenueController.js ./src/main/webapp/admin/scripts/controllers/editVenueController.js
index bc0cbeb..ac61a65 100644
--- ./src/main/webapp/admin/scripts/controllers/editVenueController.js
+++ ./src/main/webapp/admin/scripts/controllers/editVenueController.js
@@ -16,7 +16,7 @@ angular.module('ticketmonster').controller('EditVenueController', function($scop
                     };
                     var labelObject = {
                         value : item.id,
-                        text : item.mediaType
+                        text : item.url
                     };
                     if($scope.venue.mediaItem && item.id == $scope.venue.mediaItem.id) {
                         $scope.mediaItemSelection = labelObject;
@@ -103,4 +103,4 @@ angular.module('ticketmonster').controller('EditVenueController', function($scop
     });
     
     $scope.get();
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/scripts/controllers/newBookingController.js ./src/main/webapp/admin/scripts/controllers/newBookingController.js
index d68a202..4d29c29 100644
--- ./src/main/webapp/admin/scripts/controllers/newBookingController.js
+++ ./src/main/webapp/admin/scripts/controllers/newBookingController.js
@@ -27,7 +27,7 @@ angular.module('ticketmonster').controller('NewBookingController', function ($sc
         $scope.performanceSelectionList = $.map(items, function(item) {
             return ( {
                 value : item.id,
-                text : item.date
+                text : item.displayTitle
             });
         });
     });
@@ -54,4 +54,4 @@ angular.module('ticketmonster').controller('NewBookingController', function ($sc
     $scope.cancel = function() {
         $location.path("/Bookings");
     };
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/scripts/controllers/newEventController.js ./src/main/webapp/admin/scripts/controllers/newEventController.js
index 069b0c3..2cc7120 100644
--- ./src/main/webapp/admin/scripts/controllers/newEventController.js
+++ ./src/main/webapp/admin/scripts/controllers/newEventController.js
@@ -8,7 +8,7 @@ angular.module('ticketmonster').controller('NewEventController', function ($scop
         $scope.mediaItemSelectionList = $.map(items, function(item) {
             return ( {
                 value : item.id,
-                text : item.mediaType
+                text : item.url
             });
         });
     });
@@ -50,4 +50,4 @@ angular.module('ticketmonster').controller('NewEventController', function ($scop
     $scope.cancel = function() {
         $location.path("/Events");
     };
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/scripts/controllers/newPerformanceController.js ./src/main/webapp/admin/scripts/controllers/newPerformanceController.js
index 9899ee4..8ac2a3b 100644
--- ./src/main/webapp/admin/scripts/controllers/newPerformanceController.js
+++ ./src/main/webapp/admin/scripts/controllers/newPerformanceController.js
@@ -8,7 +8,7 @@ angular.module('ticketmonster').controller('NewPerformanceController', function
         $scope.showSelectionList = $.map(items, function(item) {
             return ( {
                 value : item.id,
-                text : item.id
+                text : item.displayTitle
             });
         });
     });
@@ -35,4 +35,4 @@ angular.module('ticketmonster').controller('NewPerformanceController', function
     $scope.cancel = function() {
         $location.path("/Performances");
     };
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/scripts/controllers/newSectionAllocationController.js ./src/main/webapp/admin/scripts/controllers/newSectionAllocationController.js
index 4320cb5..52a8143 100644
--- ./src/main/webapp/admin/scripts/controllers/newSectionAllocationController.js
+++ ./src/main/webapp/admin/scripts/controllers/newSectionAllocationController.js
@@ -8,7 +8,7 @@ angular.module('ticketmonster').controller('NewSectionAllocationController', fun
         $scope.performanceSelectionList = $.map(items, function(item) {
             return ( {
                 value : item.id,
-                text : item.date
+                text : item.displayTitle
             });
         });
     });
@@ -50,4 +50,4 @@ angular.module('ticketmonster').controller('NewSectionAllocationController', fun
     $scope.cancel = function() {
         $location.path("/SectionAllocations");
     };
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/scripts/controllers/newShowController.js ./src/main/webapp/admin/scripts/controllers/newShowController.js
index 01b4a50..087a23b 100644
--- ./src/main/webapp/admin/scripts/controllers/newShowController.js
+++ ./src/main/webapp/admin/scripts/controllers/newShowController.js
@@ -23,7 +23,7 @@ angular.module('ticketmonster').controller('NewShowController', function ($scope
         $scope.performancesSelectionList = $.map(items, function(item) {
             return ( {
                 value : item.id,
-                text : item.date
+                text : item.displayTitle
             });
         });
     });
@@ -57,7 +57,7 @@ angular.module('ticketmonster').controller('NewShowController', function ($scope
         $scope.ticketPricesSelectionList = $.map(items, function(item) {
             return ( {
                 value : item.id,
-                text : item.price
+                text : item.displayTitle
             });
         });
     });
@@ -88,4 +88,4 @@ angular.module('ticketmonster').controller('NewShowController', function ($scope
     $scope.cancel = function() {
         $location.path("/Shows");
     };
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/scripts/controllers/newTicketPriceController.js ./src/main/webapp/admin/scripts/controllers/newTicketPriceController.js
index daf214f..557e372 100644
--- ./src/main/webapp/admin/scripts/controllers/newTicketPriceController.js
+++ ./src/main/webapp/admin/scripts/controllers/newTicketPriceController.js
@@ -8,7 +8,7 @@ angular.module('ticketmonster').controller('NewTicketPriceController', function
         $scope.showSelectionList = $.map(items, function(item) {
             return ( {
                 value : item.id,
-                text : item.id
+                text : item.displayTitle
             });
         });
     });
@@ -16,6 +16,27 @@ angular.module('ticketmonster').controller('NewTicketPriceController', function
         if ( typeof selection != 'undefined') {
             $scope.ticketPrice.show = {};
             $scope.ticketPrice.show.id = selection.value;
+            if($scope.showList) {
+                var selectedShow = $.map($scope.showList, function(show) {
+                    if(show.id === $scope.ticketPrice.show.id) {
+                        return show;
+                    }
+                });
+                if(selectedShow.length > 0) {
+                    selectedShow = selectedShow[0];
+                }
+            }
+            if($scope.sectionList && selectedShow) {
+                var venueSectionList = $.map($scope.sectionList, function(section) {
+                    if(selectedShow.venue.id === section.venue.id) {
+                        return ( {
+                            value : section.id,
+                            text : section.name
+                        });
+                    }
+                });
+                $scope.sectionSelectionList = venueSectionList;
+            }
         }
     });
     
@@ -65,4 +86,4 @@ angular.module('ticketmonster').controller('NewTicketPriceController', function
     $scope.cancel = function() {
         $location.path("/TicketPrices");
     };
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/scripts/controllers/newVenueController.js ./src/main/webapp/admin/scripts/controllers/newVenueController.js
index 1e61887..b16736e 100644
--- ./src/main/webapp/admin/scripts/controllers/newVenueController.js
+++ ./src/main/webapp/admin/scripts/controllers/newVenueController.js
@@ -8,7 +8,7 @@ angular.module('ticketmonster').controller('NewVenueController', function ($scop
         $scope.mediaItemSelectionList = $.map(items, function(item) {
             return ( {
                 value : item.id,
-                text : item.mediaType
+                text : item.url
             });
         });
     });
@@ -54,4 +54,4 @@ angular.module('ticketmonster').controller('NewVenueController', function ($scop
     $scope.cancel = function() {
         $location.path("/Venues");
     };
-});
\ No newline at end of file
+});
diff --git ./src/main/webapp/admin/views/Booking/search.html ./src/main/webapp/admin/views/Booking/search.html
index cefb023..c0bcf5a 100644
--- ./src/main/webapp/admin/views/Booking/search.html
+++ ./src/main/webapp/admin/views/Booking/search.html
@@ -37,7 +37,7 @@
     <div class="form-group">
         <label for="performance" class="col-sm-2 control-label">Performance</label>
         <div class="col-sm-10">
-            <select id="performance" name="performance" class="form-control" ng-model="search.performance" ng-options="p as p.date for p in performanceList">
+            <select id="performance" name="performance" class="form-control" ng-model="search.performance" ng-options="p as p.displayTitle for p in performanceList">
                 <option value="">Choose a Performance</option>
             </select>
         </div>
@@ -87,4 +87,4 @@
         </li>
     </ul>
 
-</div>
\ No newline at end of file
+</div>
diff --git ./src/main/webapp/admin/views/Event/detail.html ./src/main/webapp/admin/views/Event/detail.html
index 78d55b3..4da1871 100644
--- ./src/main/webapp/admin/views/Event/detail.html
+++ ./src/main/webapp/admin/views/Event/detail.html
@@ -23,6 +23,8 @@
         <select id="mediaItem" name="mediaItem" class="form-control" ng-model="mediaItemSelection" ng-options="m.text for m in mediaItemSelectionList"  >
             <option value="">Choose a Media Item</option>
         </select>
+        <br/>
+        <img class="img-thumbnail col-md4" ng-hide="!mediaItemSelection.text" ng-src="{{mediaItemSelection.text}}" />
         </div>
     </div>
 
@@ -54,4 +56,4 @@
             <button id="deleteEvent" name="deleteEvent" class="btn btn-danger" ng-show="event.id" ng-click="remove()"><span class="glyphicon glyphicon-trash"></span> Delete</button>
         </div>
     </div>
-</form>
\ No newline at end of file
+</form>
diff --git ./src/main/webapp/admin/views/Event/search.html ./src/main/webapp/admin/views/Event/search.html
index c37f281..a2f614e 100644
--- ./src/main/webapp/admin/views/Event/search.html
+++ ./src/main/webapp/admin/views/Event/search.html
@@ -21,7 +21,7 @@
     <div class="form-group">
         <label for="mediaItem" class="col-sm-2 control-label">Media Item</label>
         <div class="col-sm-10">
-            <select id="mediaItem" name="mediaItem" class="form-control" ng-model="search.mediaItem" ng-options="m as m.mediaType for m in mediaItemList">
+            <select id="mediaItem" name="mediaItem" class="form-control" ng-model="search.mediaItem" ng-options="m as m.url for m in mediaItemList">
                 <option value="">Choose a Media Item</option>
             </select>
         </div>
@@ -65,7 +65,7 @@
             <tbody id="search-results-body">
                 <tr ng-repeat="result in searchResults | searchFilter:searchResults | startFrom:currentPage*pageSize | limitTo:pageSize">
                     <td><a href="#/Events/edit/{{result.id}}">{{result.name}}</a></td>
-                        <td><a href="#/Events/edit/{{result.id}}">{{result.mediaItem.mediaType}}</a></td>
+                        <td><a href="#/Events/edit/{{result.id}}">{{result.mediaItem.url}}</a></td>
                         <td><a href="#/Events/edit/{{result.id}}">{{result.category.description}}</a></td>
                     <td><a href="#/Events/edit/{{result.id}}">{{result.description}}</a></td>
                 </tr>
@@ -85,4 +85,4 @@
         </li>
     </ul>
 
-</div>
\ No newline at end of file
+</div>
diff --git ./src/main/webapp/admin/views/Performance/search.html ./src/main/webapp/admin/views/Performance/search.html
index ef13770..b315edf 100644
--- ./src/main/webapp/admin/views/Performance/search.html
+++ ./src/main/webapp/admin/views/Performance/search.html
@@ -14,7 +14,7 @@
     <div class="form-group">
         <label for="show" class="col-sm-2 control-label">Show</label>
         <div class="col-sm-10">
-            <select id="show" name="show" class="form-control" ng-model="search.show" ng-options="s as s.id for s in showList">
+            <select id="show" name="show" class="form-control" ng-model="search.show" ng-options="s as s.displayTitle for s in showList">
                 <option value="">Choose a Show</option>
             </select>
         </div>
@@ -40,7 +40,7 @@
             </thead>
             <tbody id="search-results-body">
                 <tr ng-repeat="result in searchResults | searchFilter:searchResults | startFrom:currentPage*pageSize | limitTo:pageSize">
-                        <td><a href="#/Performances/edit/{{result.id}}">{{result.show.id}}</a></td>
+                        <td><a href="#/Performances/edit/{{result.id}}">{{result.show.displayTitle}}</a></td>
                     <td><a href="#/Performances/edit/{{result.id}}">{{result.date| date:'yyyy-MM-dd HH:mm:ss Z'}}</a></td>
                 </tr>
             </tbody>
@@ -59,4 +59,4 @@
         </li>
     </ul>
 
-</div>
\ No newline at end of file
+</div>
diff --git ./src/main/webapp/admin/views/SectionAllocation/search.html ./src/main/webapp/admin/views/SectionAllocation/search.html
index 820fdd5..b813ea8 100644
--- ./src/main/webapp/admin/views/SectionAllocation/search.html
+++ ./src/main/webapp/admin/views/SectionAllocation/search.html
@@ -20,7 +20,7 @@
     <div class="form-group">
         <label for="performance" class="col-sm-2 control-label">Performance</label>
         <div class="col-sm-10">
-            <select id="performance" name="performance" class="form-control" ng-model="search.performance" ng-options="p as p.date for p in performanceList">
+            <select id="performance" name="performance" class="form-control" ng-model="search.performance" ng-options="p as p.displayTitle for p in performanceList">
                 <option value="">Choose a Performance</option>
             </select>
         </div>
@@ -57,7 +57,7 @@
             <tbody id="search-results-body">
                 <tr ng-repeat="result in searchResults | searchFilter:searchResults | startFrom:currentPage*pageSize | limitTo:pageSize">
                     <td><a href="#/SectionAllocations/edit/{{result.id}}">{{result.occupiedCount}}</a></td>
-                        <td><a href="#/SectionAllocations/edit/{{result.id}}">{{result.performance.date| date:'medium'}}</a></td>
+                        <td><a href="#/SectionAllocations/edit/{{result.id}}">{{result.performance.displayTitle}}</a></td>
                         <td><a href="#/SectionAllocations/edit/{{result.id}}">{{result.section.name}}</a></td>
                 </tr>
             </tbody>
@@ -76,4 +76,4 @@
         </li>
     </ul>
 
-</div>
\ No newline at end of file
+</div>
diff --git ./src/main/webapp/admin/views/TicketPrice/search.html ./src/main/webapp/admin/views/TicketPrice/search.html
index 2bf2648..4345e63 100644
--- ./src/main/webapp/admin/views/TicketPrice/search.html
+++ ./src/main/webapp/admin/views/TicketPrice/search.html
@@ -14,7 +14,7 @@
     <div class="form-group">
         <label for="show" class="col-sm-2 control-label">Show</label>
         <div class="col-sm-10">
-            <select id="show" name="show" class="form-control" ng-model="search.show" ng-options="s as s.id for s in showList">
+            <select id="show" name="show" class="form-control" ng-model="search.show" ng-options="s as s.displayTitle for s in showList">
                 <option value="">Choose a Show</option>
             </select>
         </div>
@@ -66,7 +66,7 @@
             </thead>
             <tbody id="search-results-body">
                 <tr ng-repeat="result in searchResults | searchFilter:searchResults | startFrom:currentPage*pageSize | limitTo:pageSize">
-                        <td><a href="#/TicketPrices/edit/{{result.id}}">{{result.show.id}}</a></td>
+                        <td><a href="#/TicketPrices/edit/{{result.id}}">{{result.show.displayTitle}}</a></td>
                         <td><a href="#/TicketPrices/edit/{{result.id}}">{{result.section.name}}</a></td>
                         <td><a href="#/TicketPrices/edit/{{result.id}}">{{result.ticketCategory.description}}</a></td>
                     <td><a href="#/TicketPrices/edit/{{result.id}}">{{result.price}}</a></td>
@@ -87,4 +87,4 @@
         </li>
     </ul>
 
-</div>
\ No newline at end of file
+</div>
diff --git ./src/main/webapp/admin/views/Venue/detail.html ./src/main/webapp/admin/views/Venue/detail.html
index 5bb6383..77904e1 100644
--- ./src/main/webapp/admin/views/Venue/detail.html
+++ ./src/main/webapp/admin/views/Venue/detail.html
@@ -20,6 +20,8 @@
         <select id="mediaItem" name="mediaItem" class="form-control" ng-model="mediaItemSelection" ng-options="m.text for m in mediaItemSelectionList"  >
             <option value="">Choose a Media Item</option>
         </select>
+        <br/>
+        <img class="img-thumbnail col-md4" ng-hide="!mediaItemSelection.text" ng-src="{{mediaItemSelection.text}}" />
         </div>
     </div>
 
@@ -76,4 +78,4 @@
             <button id="deleteVenue" name="deleteVenue" class="btn btn-danger" ng-show="venue.id" ng-click="remove()"><span class="glyphicon glyphicon-trash"></span> Delete</button>
         </div>
     </div>
-</form>
\ No newline at end of file
+</form>
diff --git ./src/main/webapp/admin/views/Venue/search.html ./src/main/webapp/admin/views/Venue/search.html
index 57bc1ff..ff04b44 100644
--- ./src/main/webapp/admin/views/Venue/search.html
+++ ./src/main/webapp/admin/views/Venue/search.html
@@ -21,7 +21,7 @@
     <div class="form-group">
         <label for="mediaItem" class="col-sm-2 control-label">Media Item</label>
         <div class="col-sm-10">
-            <select id="mediaItem" name="mediaItem" class="form-control" ng-model="search.mediaItem" ng-options="m as m.mediaType for m in mediaItemList">
+            <select id="mediaItem" name="mediaItem" class="form-control" ng-model="search.mediaItem" ng-options="m as m.url for m in mediaItemList">
                 <option value="">Choose a Media Item</option>
             </select>
         </div>
@@ -88,7 +88,7 @@
             <tbody id="search-results-body">
                 <tr ng-repeat="result in searchResults | searchFilter:searchResults | startFrom:currentPage*pageSize | limitTo:pageSize">
                     <td><a href="#/Venues/edit/{{result.id}}">{{result.name}}</a></td>
-                        <td><a href="#/Venues/edit/{{result.id}}">{{result.mediaItem.mediaType}}</a></td>
+                        <td><a href="#/Venues/edit/{{result.id}}">{{result.mediaItem.url}}</a></td>
                     <td><a href="#/Venues/edit/{{result.id}}">{{result.description}}</a></td>
                     <td><a href="#/Venues/edit/{{result.id}}">{{result.capacity}}</a></td>
                     <td><a href="#/Venues/edit/{{result.id}}">{{result.address.street}}</a></td>
@@ -111,4 +111,4 @@
         </li>
     </ul>
 
-</div>
\ No newline at end of file
+</div>
